### ISSUE TITLE
Custom namespace prefixing

### ISSUE THREAD

**Issue by dignifiedquire** [getditto/safer_ffi#93]
I was wondering if there is a way to automatically add a prefix to types/functions etc when generating the C definitions, sth like this:

```rust
#[ffi_export]
fn foo() {
  // ..things
}

#[derive_reprC]
pub struct MyStruct { /* ...  */ }
```

```c
typedef struct { /* .. */ } PFX_MyStruct;
void pfx_foo() {
}
```

as currently I have to add the `PFX` prefix manually to my types and functions in rust, which makes it very unidiomatic, and easy to forget.

**Comment by danielhenrymantilla** [comment]
Agreed. This more general query thus supersedes https://github.com/getditto/safer_ffi/issues/70, and ought to be featured rather soon 🙂

**Comment by danielhenrymantilla** [comment]
Yes and no: I haven't had the time to cater to the finishing touches for the header generation, which I would like to make it as extensible as possible, with a guide about this, _c.f._ the `languages/{c,csharp}.rs` files; so some certain parts are currently empty/shallow scaffolding for when they finally do something in the future.
It is, however, a genuine oversight for this scaffolding to have ended up in public API, such as `with_naming_convention()` 😅

**Comment by UpcraftLP** [comment]
Any updates on this?

being able to automatically prefix generated function names would be very helpful indeed
